apps/aechannel/src/aesc_fsm.erl

```diff
@@ -1257,12 +1330,47 @@ mutual_closing(cast, {closing_signed, _Msg}, D) ->  %% TODO: not using this, rig
 mutual_closing(Type, Msg, D) ->
     handle_common_event(Type, Msg, error, D).
 
-channel_closing(enter, _OldSt, _D) -> keep_state_and_data;
+channel_closing(enter, _OldSt, D) ->
+    D1 = if D#data.channel_status =/= closing ->
+                 report(info, closing, D),
+                 report_update(D#data{channel_status = closing});
+            true ->
+                 report_update(D)
+         end,
+    keep_state(D1#data{ongoing_update = false});
+channel_closing(cast, {?CHANNEL_UNLOCKED, #{chan_id := ChId}} = Msg,
+                #data{on_chain_id = ChId, latest = {watch, unlock, _TxHash, SignedTx, _}} = D) ->
+    lager:debug("channel unlocked", []),
+    D1 = log(rcv, ?CHANNEL_UNLOCKED, Msg, D),
+    {Type, _Tx} = aetx:specialize_type(aetx_sign:tx(SignedTx)),
+    case Type of
+        channel_settle_tx ->
+            lager:debug("pushing settle tx", []),
+            ok = aec_tx_pool:push(SignedTx);
+        _ ->
+            lager:debug("SignedTx of Type ~p - ignoring", [Type]),
+            ok
+    end,
+    keep_state(D1);
 channel_closing(Type, Msg, D) ->
     handle_common_event(Type, Msg, discard, D).
 
+channel_closed(enter, _OldSt, D) ->
+    keep_state(D#data{ongoing_update = false});
+channel_closed(cast, {?MIN_DEPTH_ACHIEVED, ChainId, ?WATCH_CLOSED, TxHash},
+               #data{ latest = {?MIN_DEPTH, ?WATCH_CLOSED, TxHash, _, _}
+                    , on_chain_id = ChainId } = D) ->
+    close(closed_confirmed, D);
+channel_closed(cast, {?CHANNEL_CHANGED, _Info} = Msg, D) ->
+    %% This is a weird case. The channel was closed, but now isn't.
+    %% This would indicate a fork switch. TODO: detect channel status
+    %% and respond accordingly. For now. Panic and terminate
+    report(info, zombie_channel, D),
+    close(zombie_channel, log(rcv, msg_type(Msg), Msg, D));
```

Comment:
This is already tracked as [Handle zombie state channels](https://www.pivotaltracker.com/story/show/165438326). Ok